hazelcast/include/hazelcast/client/internal/socket/BaseSocket.h

```diff
@@ -90,54 +90,45 @@ namespace hazelcast {
                                     const std::shared_ptr<spi::impl::ClientInvocation> invocation) override {
                         check_connection(connection, invocation);
                         auto message = invocation->get_client_message();
-                        socket_strand_.post([=]() {
+                        socket_strand_.post([connection, invocation, message, this]() {
```

Comment:
Why this change is required? What is difference between [=]  and [variables]?

Reply:
This makes the variables used more explicit. Otherwise, the other version is implicit and you can easily be coping variables you did not intend copy in the first place.  We switched to this usage after getting more experience with lambdas.